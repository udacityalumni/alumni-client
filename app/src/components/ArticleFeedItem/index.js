import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Image from 'grommet-udacity/components/Image';
import Markdown from 'grommet-udacity/components/Markdown';
import Article from 'grommet-udacity/components/Article';
import Heading from 'grommet-udacity/components/Heading';
import Anchor from 'grommet-udacity/components/Anchor';
import Box from 'grommet-udacity/components/Box';
import Label from 'grommet-udacity/components/Label';

const ArticleFeedItem = ({
  article,
}) => (
  <Article className={styles.articleFeedItem}>
    <Box>
      {article.tags && article.tags.map((tag, i) =>
        <Anchor
          key={i}
          label={tag.tag}
          className={styles.tagLink}
          href={`/tags/${tag.tag}`}
        />
    ).join(', ')}
    </Box>
    <Heading strong align="start" tag="h2">
      {article.title.slice(0, 30)}
    </Heading>
    {article.feature_image &&
      <Image src={article.feature_image} />
    }
    <Markdown content={`${article.content.slice(0, 350)}...`} />
    <Box align="start" direction="row" justify="center">
      <Anchor
        href={`/articles/${article.id}`}
        primary
        label="Read More"
      />
      <div style={{ flex: 1 }}>
        <Label uppercase>
          - {article.user.name}
        </Label>
      </div>
    </Box>
  </Article>
);

ArticleFeedItem.propTypes = {

};

export default cssModules(ArticleFeedItem, styles);
