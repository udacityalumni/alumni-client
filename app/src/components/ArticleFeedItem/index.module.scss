.tagLink {
  color: #7D97AD;
  text-transform: uppercase;
  font-size: 11px;
  letter-spacing: 0.1em;
}
