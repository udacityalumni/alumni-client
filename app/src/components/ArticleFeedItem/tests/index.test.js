import ArticleFeedItem from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import article from './mocks';

describe('<ArticleFeedItem />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ArticleFeedItem
        article={article}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
