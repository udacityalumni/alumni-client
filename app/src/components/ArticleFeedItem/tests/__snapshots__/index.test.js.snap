exports[`<ArticleFeedItem /> should render with default props 1`] = `
<Article
  direction="column"
  pad="none">
  <Heading
    tag="h1">
    Administrative services only H
  </Heading>
  <Image
    size="medium"
    src="https://robohash.org/dignissimosased.png?size=300x300" />
  <Markdown
    components={Object {}}
    content="Skilled nursing facility lifetime maximum special benefit networks qualifying event short-term medical insurance. Hedis CPT referral guaranteed issue pre-certification. Network provider HEDIS medicare coordination of benefits any willing provider laws gatekeeper evidence of insurability. Network pro" />
  <Anchor
    href="/articles/3"
    label="Read More"
    primary={true}
    tag="a" />
</Article>
`;
